packages/flutter/test/cupertino/route_test.dart

```diff
@@ -1151,3 +1183,39 @@ class DialogObserver extends NavigatorObserver {
     super.didPush(route, previousRoute);
   }
 }
+
+void checkPreviousPageTitle(String previousPageTitle, WidgetTester tester) {
+  // Also shows the previous page's title next to the back button.
+  expect(find.widgetWithText(CupertinoButton, previousPageTitle), findsOneWidget);
+  // 2 paddings + 1 ahem character at font size 34.0.
+  expect(tester.getTopLeft(find.text(previousPageTitle)).dx, 8.0 + 34.0 + 6.0);
+}
+
+void checkCurrentPageTitle(String currentPageTitle, WidgetTester tester) {
+  expect(find.widgetWithText(CupertinoNavigationBar, currentPageTitle), findsOneWidget);
+  expect(tester.getCenter(find.text(currentPageTitle)).dx, 400.0);
+}
+
+CupertinoPageRoute<void> buildCupertinoPageRouteWithNavBar(String titleText, {RouteSettings settings = const RouteSettings()}) =>
+    CupertinoPageRoute<void>(
+      title: titleText,
+      settings: settings,
+      builder: (BuildContext context) {
+        return const CupertinoPageScaffold(
+          navigationBar: CupertinoNavigationBar(),
+          child: Placeholder(),
+        );
+      },
+    );
+
+const String PAGE_1 = 'page_1',
+    PAGE_2 = 'page_2',
+    PAGE_3 = 'page_3',
+    PAGE_4 = 'page_4',
+    PAGE_5 = 'page_5',
+    PAGE_6 = 'page_6';
+
+Future<void> pumpAndWait(WidgetTester tester) async{
```

Comment:
there are still some tests uses 
```dart
await tester.pump();
await tester.pump(const Duration(milliseconds: 500));
```
Can you also refactor them to use this method?